frontend/src/store/user-details/reducer.ts

```diff
@@ -0,0 +1,57 @@
+import { createReducer } from '@reduxjs/toolkit';
+import { DataStatus } from 'common/enums/enums';
+import { UserDetailsResponseDto } from 'common/types/types';
+
+import {
+  getUserDetails,
+  updateUserDetails,
+  updateUserDetailsAvatar,
+} from './actions';
+
+type State = {
+  dataStatus: DataStatus;
+  userDetails: UserDetailsResponseDto | null;
+  avatarUrl: string | null;
+};
+
+const initialState: State = {
+  dataStatus: DataStatus.IDLE,
+  userDetails: null,
+  avatarUrl: null,
+};
+
+const reducer = createReducer(initialState, (builder) => {
+  builder.addCase(getUserDetails.pending, (state) => {
+    state.dataStatus = DataStatus.PENDING;
+  });
+  builder.addCase(getUserDetails.fulfilled, (state, action) => {
+    state.dataStatus = DataStatus.FULFILLED;
+    state.userDetails = action.payload;
+    state.avatarUrl = action.payload.avatarUrl;
```

Comment:
On BE `getByUserId` method in `UserDetailsService` may return `null` and it could cause some failures in case when response is `null`

Reply:
actually in this case UserDetailsService doesn't return null, put method on backend for updating the avatar returns body always with field avatarUrl. So, didn't get this comment, maybe I missed something?